File: backend/app/chat_commands.py
import re

_ZERO_WIDTH = re.compile(r"[\u200b-\u200d\ufeff]")


def is_clear_chat_command(message: str) -> bool:
    normalized = _ZERO_WIDTH.sub("", message.strip()).replace("\uff0f", "/").lower()
    normalized = re.sub(r"[.!?]+$", "", normalized)
    if normalized.startswith("/"):
        normalized = normalized[1:]
    return normalized == "clear"

File: frontend/src/lib/chatCommands.ts
const ZERO_WIDTH_CHARS = /[\u200B-\u200D\uFEFF]/g;

/** Normalize user input for slash-command matching. */
export function normalizeChatCommandInput(raw: string): string {
  return raw
    .trim()
    .replace(ZERO_WIDTH_CHARS, '')
    .replace(/\uFF0F/g, '/')
    .toLowerCase()
    .replace(/[.!?]+$/, '');
}

/** True when the user intends to reset the chat (e.g. /clear, clear). */
export function isClearChatCommand(raw: string): boolean {
  const normalized = normalizeChatCommandInput(raw);
  return normalized === '/clear' || normalized === 'clear';
}
